library/core/src/slice/mod.rs

```diff
@@ -1099,10 +1099,16 @@ impl<T> [T] {
     /// assert!(iter.next().is_none());
     /// ```
     ///
-    /// There's no `windows_mut`, as that existing would let safe code violate the
-    /// "only one `&mut` at a time to the same thing" rule.  However, you can sometimes
-    /// use [`Cell::as_slice_of_cells`](crate::cell::Cell::as_slice_of_cells) in
-    /// conjunction with `windows` to accomplish something similar:
+    /// Because the [Iterator] trait cannot represent the required lifetimes,
+    /// there is no `windows_mut` analog to `windows`;
+    /// `[0,1,2].windows_mut(2).collect()` would violate [the rules of references]
+    /// (though a [LendingIterator] analog is possible).
+    /// You can sometimes use
+    /// [`Cell::as_slice_of_cells`](crate::cell::Cell::as_slice_of_cells) in
+    /// conjunction with `windows` instead:
```

Comment:
```suggestion
    /// Because the [Iterator] trait cannot represent the required lifetimes, there is no
	/// `windows_mut` analog to `windows`; `[0,1,2].windows_mut(2).collect()` would violate
	/// [the rules of references] (though a [LendingIterator] analog is possible). You can sometimes
	/// use [`Cell::as_slice_of_cells`](crate::cell::Cell::as_slice_of_cells) in conjunction with
    /// `windows` instead:
```

Looks like it may have gotten chopped up somehow, it should be something like the wrapping here (I wish rustfmt just did this...)

No worries about links, those are find to exceed 100.

Reply:
I did manual line breaking at logical places, which is why it looks a bit ragged.

Reply:
Ah, makes sense. We tend to avoid semantic line breaks in docs because it's a small benefit for reading diffs but makes reading the code less nice, and the code gets read significantly more than diffs do.

Anyway, thanks for all the follow up here!